smart-contracts/advanced/deterministic-deployment.md

```diff
@@ -0,0 +1,141 @@
+## What is deterministic deployment?
+
+Deterministic deployment refers to the ability to deploy smart contracts to the same address across different blockchain networks. This consistency can be particularly useful when deploying the same contract on multiple chains to streamline interactions or ensure interoperability. There are two primary methods to achieve deterministic deployment:
+
+- Using a **factory contract that has been deployed keylessly** to deploy your contracts.
+    - The factory usually uses the `CREATE2` opcode to deploy your contracts, precomputing the deployment address and ensuring consistency across all chains.
+- **Deploying your contracts keylessly** (without using any factory) onto all blockchains.
+    - You must keep the code and constructor arguments unchanged to get the same address.
+
+## Creating a deterministic deployment
+
+Contract address can be precomputed, before the contract is deployed, using a create opcode. This is one of the options for deterministic deployment.
+
+### **`CREATE`**
+
+If deploying from an externally-owned account (EOA), e.g. by calling `ethers.deployContract` from a hardhat script, the `CREATE` opcode will run in the EVM and contract addresses will be calculated using the:
```

Comment:
Based on comment above, do we need this section about deploying from EOA? We can keep it for reference purposes to show difference between deploying from EOA VS having a factory contract but would add a line that this is not the focus of the article

Reply:
Deleted